Add asterisks to required fields

KarolDawidziuk opened the pull request:
<!--
🚨 If you want to submit a PR for a security vulnerability, please contact us directly
at https://ckeditor.com/contact/ instead. 🚨
-->
## What is the purpose of this pull request?

New feature

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [ ] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [x] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

```
* [#3433>](https://github.com/ckeditor/ckeditor4/issues/3433): [dialog]: Added asterisks to required fields.
```

## What changes did you make?

I've added asterisks to each required label in the dialogs.

## Which issues does your PR resolve?

Closes #3433.

jacekbogdanski commented on tests/plugins/dialogui/required.js:
```diff
@@ -0,0 +1,97 @@
+							label = document.getElementById( domId ),
+							labelContent = CKEDITOR.env.safari ? label.textContent : label.innerText;
+
+						assert.areSame( 'bar*', labelContent );
```
It would be good to rather check HTML instead of text, so we can confirm that the asterisk has correct styling and structure.

jacekbogdanski commented on plugins/dialogui/plugin.js:
```diff
@@ -151,7 +151,8 @@ CKEDITOR.plugins.add( 'dialogui', {
 									( elementDefinition.labelStyle ? ' style="' + elementDefinition.labelStyle + '"' : '' ) + '>',
-									elementDefinition.label,
+									// If label is required add red asterisk. (#3433)
+									( elementDefinition.required ? elementDefinition.label + '<span style="color:red;">*</span>' : elementDefinition.label ),
```
Instead of inlining styling, we could make this feature a bit more extensible and add CSS class instead, like `cke_dialog_ui_labeled_required`. Such class should be defined in our `./skins` (you can check out how `cke_dialog_ui_labeled_label` was defined), so the option is customizable.

jacekbogdanski reviewed (changes requested):
Please, add a manual test confirming that the feature works correctly for different editor skins (required once you move inlined CSS). You can just use some existing feature dialog like image, table, etc

KarolDawidziuk commented:
Thanks @jacekbogdanski  for the review! 
It's ready for another one.

Changes: 
*  Added custom CSS class instead of inline style.
* Updated unit tests.
* Added manual tests.

jacekbogdanski commented on plugins/dialogui/plugin.js:
```diff
@@ -151,7 +151,8 @@ CKEDITOR.plugins.add( 'dialogui', {
 									( elementDefinition.labelStyle ? ' style="' + elementDefinition.labelStyle + '"' : '' ) + '>',
-									elementDefinition.label,
+									// If label is required add red asterisk. #3433)
+									( elementDefinition.required ? elementDefinition.label + '<span class="cke_dialog_ui_labeled_required" >*</span>' : elementDefinition.label ),
```
```suggestion
									( elementDefinition.required ? elementDefinition.label + '<span class="cke_dialog_ui_labeled_required">*</span>' : elementDefinition.label ),
```

jacekbogdanski commented on skins/kama/dialog.css:
```diff
@@ -352,6 +352,11 @@ div.cke_disabled .cke_dialog_ui_labeled_content *
 }
 
+/* (#3433) */
+.cke_dialog_ui_labeled_required{
+	color: red;
+}
```
Let's stick to the file styling.
```suggestion
.cke_dialog_ui_labeled_required
{
	color: red;
}
```

jacekbogdanski commented on skins/moono-lisa/dialog.css:
```diff
@@ -752,6 +752,11 @@ select.cke_dialog_ui_input_select
 	margin-left: 1px;
 }
 
+/* (#3433) */
+.cke_dialog_ui_labeled_required{
+	color: red;
+}
```
Let's stick to the file styling.
```suggestion
/* (#3433) */
.cke_dialog_ui_labeled_required
{
	color: red;
}
```

jacekbogdanski commented on skins/moono/dialog.css:
```diff
@@ -399,6 +399,11 @@ div.cke_disabled .cke_dialog_ui_labeled_content div *
 	cursor: default;
 }
 
+/* (#3433) */
+.cke_dialog_ui_labeled_required{
+	color: red;
+}
```
Let's stick to the file styling.
```suggestion
/* (#3433) */
.cke_dialog_ui_labeled_required
{
	color: red;
}
```

jacekbogdanski reviewed (approved):
Good job 👍🏻
